appserver/microprofile/health-glassfish/src/main/java/org/glassfish/microprofile/impl/HealthService.java

```diff
@@ -0,0 +1,31 @@
+package org.glassfish.microprofile.impl;
+
+
+import org.glassfish.api.StartupRunLevel;
+import org.glassfish.api.event.EventListener;
+import org.glassfish.hk2.runlevel.RunLevel;
+import org.glassfish.internal.api.Globals;
+import org.glassfish.internal.data.ApplicationInfo;
+import org.glassfish.internal.deployment.Deployment;
+import org.glassfish.microprofile.health.HealthReporter;
+import org.jvnet.hk2.annotations.Service;
+
+@Service(name = "healthcheck-service")
+@RunLevel(StartupRunLevel.VAL)
+public class HealthService implements EventListener {
+
+
+    @Override
+    public void event(Event<?> event) {
+
+        HealthReporter service = Globals.getDefaultHabitat().getService(HealthReporter.class);
+
+        if (event.is(Deployment.APPLICATION_UNLOADED) && event.hook() instanceof ApplicationInfo appInfo) {
+            service.removeAllHealthChecksFrom(appInfo.getName());
```

Comment:
The healthchecks are registered with "" context root but here we want to remove them by `appInfo.getName()`, which is not "". As a result, the health checks are not removed on undeployment, remain leaking, and even remain in the output of the `/health` endpoint.